### PR TITLE
Bypass upgrade page when occ controller is requested

### PR DESCRIPTION
**Pull request by VicDeo**
Request to `domain.tld/index.php/occ/upgrade` respond with a `pending upgrade` page otherwise

To test:
1. increase oc version in `version.php`or any app version
2. try to access `domain.tld/index.php/occ/upgrade` 

Expected: Response is json with the respective command result

Actual: Response is  `Your ownCloud is going to be upgraded`  HTML page

### CODE REVIEW

**Comment by PVince81** [comment]
@VicDeo how come this worked on 9.1 but not 9.0 ? Did we miss a backport or is this piece of code different ?

**Comment by PVince81** [comment]
Ah I see, was broken on master too.

File: lib/base.php [hunk 1]
```diff
@@ -831,10 +831,14 @@ public static function handleRequest() {
 		}
 
 		$request = \OC::$server->getRequest();
+		// Check if requested URL matches 'index.php/occ'
+		$isOccControllerRequested = preg_match('|/index\.php$|', $request->getScriptName()) === 1
```
Comment (PVince81) [inline lib/base.php]: shouldn't occ be part of the regexp?

Comment (PVince81) [inline lib/base.php]: never mind, I jsut saw "getScriptName" vs "getPathInfo"

**Comment by PVince81** [comment]
I was on a stable9-related branch then switched to this one. An update is due.

I opened http://localhost/owncloud/index.php/occ/upgrade in the web browser but am getting an error page "Internal Server Error"
```
{"reqId":"UvYeYjjwbzT38z8Ev62l","remoteAddr":"127.0.0.1","app":"index","message":"Exception: {\"Exception\":\"Symfony\\\\Component\\\\Routing\\\\Exception\\\\MethodNotAllowedException\",\"Message\":\"\",\"Code\":0,\"Trace\":\"#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(256): Symfony\\\\Component\\\\Routing\\\\Matcher\\\\UrlMatcher->match('\\\/occ\\\/upgrade')\\n#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/base.php(864): OC\\\\Route\\\\Router->match('\\\/occ\\\/upgrade')\\n#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/index.php(39): OC::handleRequest()\\n#3 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/symfony\\\/routing\\\/Matcher\\\/UrlMatcher.php\",\"Line\":97}","level":3,"time":"2016-07-05T09:25:58+00:00","method":"GET","url":"\/owncloud\/index.php\/occ\/upgrade","user":"--"}
```

@VicDeo Which method to use for that URL ?

**Comment by PVince81** [comment]
Steps:

1. Check out stable9 (fdb0d4ad528425b934d9b039c9f09c132b86d0f4)
1. Switch to this branch
1. Open web browser and see pending update
1. Open http://localhost/owncloud/index.php/occ/upgrade in web browser

Getting the error from above.

**Comment by VicDeo** [comment]
@PVince81 POST https://github.com/owncloud/core/blob/master/core/routes.php#L51

**Comment by PVince81** [comment]
Tested, works :+1:

**Comment by PVince81** [comment]
Second review please @DeepDiver1975 @georgehrke

**Comment by georgehrke** [comment]
👍

**Comment by PVince81** [comment]
@VicDeo please submit the backport PR for stable9.
I already tested it when patching for https://github.com/owncloud/updater/pull/355 and can confirm it works.

**Comment by VicDeo** [comment]
Stable9: https://github.com/owncloud/core/pull/25363

@PVince81 
and I guess stable9.1 needs this too

**Pull request merged**